### PR TITLE
[Queues] Update tutorial on Browser Rendering

### PR DESCRIPTION
**Pull request by kodster28**
### Summary

This tutorial ranks surprisingly high for our Developer platform tutorials for traffic and search volume (18 / 56), but it hasn't been touched in over a year.

Additionally, felt like there were some clear improvements such as including unacknowledged dependencies, clearing up the steps, and including the full working script at the end (I had to grab it from the GitHub repo myself).

While I was doing this, also moved a few Queues things to partials so we could share them across the get started guide and this tutorial.

Can't exactly speak to the code quality, but the output appeared to work for me.

### CODE REVIEW

File: content/queues/tutorials/web-crawler-with-browser-rendering.md [hunk 1]
```diff
@@ -1,116 +1,184 @@
...
+main = "src/index.ts"
+compatibility_date = "2024-07-25"
+compatibility_flags = ["nodejs_compat"]
+
+kv_namespaces = [
+  { binding = "CRAWLER_SCREENSHOTS_KV", id = "<GENERATED_NAMESPACE_ID>" },
+  { binding = "CRAWLER_LINKS_KV", id = "<GENERATED_NAMESPACE_ID>" }
+]
+
+browser = { binding = "CRAWLER_BROWSER" }
```
Comment (kodster28) [inline content/queues/tutorials/web-crawler-with-browser-rendering.md]: Might be ignorance on my end, but couldn't figure out how to either re-format the queues to a JSON format OR get browser rendering to match the `[[browser]]` version.

File: content/queues/tutorials/web-crawler-with-browser-rendering.md [hunk 2]
```diff
@@ -0,0 +1,583 @@
...
+title: Build a web crawler with Queues and Browser Rendering
+summary: Example of how to use Queues and Browser Rendering to power a web crawler.
+content_type:  📝 Tutorial
+pcx_content_type: tutorial
+products: [Browser Rendering, KV]
+---
+
+# Build a web crawler with Queues and Browser Rendering
+
+{{<tutorial-date-info>}}
```
Comment (kodster28) [inline content/queues/tutorials/web-crawler-with-browser-rendering.md]: Diff appears larger than before b/c I moved the file from a folder to a regular .md file.

**Comment by kodster28** [comment]
> This PR changes current filenames or deletes current files. Make sure you have [redirects](https://developers.cloudflare.com/pages/configuration/redirects/) set up to cover the following paths:
> 
> * [x]  `/queues/tutorials/web-crawler-with-browser-rendering/`

This isn't necessary b/c the URL path stays the same (I just moved from a folder to a file)

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md [hunk 3]
```diff
@@ -1,60 +1,36 @@
...
 
 # Build a web crawler with Queues and Browser Rendering
 
 {{<tutorial-date-info>}}
 
-This tutorial explains how to build and deploy a web crawler with Queues, Browser Rendering, and Puppeteer.
+This tutorial explains how to build and deploy a web crawler with Queues, [Browser Rendering](/browser-rendering/), and [Puppeteer](/browser-rendering/platform/puppeteer/).
 
-Puppeteer is a high-level library used to automate interactions with Chrome/Chromium browsers. On each submitted page, the crawler will find the number of links to Cloudflare.com and take a screenshot of the site, saving results to KV.
+Puppeteer is a high-level library used to automate interactions with Chrome/Chromium browsers. On each submitted page, the crawler will find the number of links to `cloudflare.com` and take a screenshot of the site, saving results to KV as an [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
```
Comment (maheshwarip) [inline content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: IMO we don't need to specify that we're saving the results as an ArrayBuffer, we can just leave it at "saving the results to KV". Also: should it be `Workers KV`, not `KV`? And should we link out to https://developers.cloudflare.com/kv/?

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md [hunk 4]
```diff
@@ -66,47 +42,134 @@ Then, move into your newly created directory:
...
-In the `wrangler.toml` file, add a Browser Rendering binding. Adding a Browser Rendering binding gives the Worker access to a headless Chromium instance you will control with Puppeteer. Add the bindings for KV namespaces and queue you created previously, which will allow you to access KV and the queue from the Worker.
+We need to create a KV store. This can be done through the Cloudflare dashboard or the Wrangler CLI. For this tutorial, we will use the Wrangler CLI.
 
-```toml
+```sh
+$ npx wrangler kv namespace create crawler_links
+$ npx wrangler kv namespace create crawler_screenshots
+```
+
+The `wrangler kv namespace create assets` subcommand creates a KV namespace by concatenating your Worker's name and the value provided for `assets`. An `id` will be randomly generated for the KV namespace.
```
Comment (maheshwarip) [inline content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: what is this referring to? I got lost when I read this! typo perhaps, or copy-paste error?

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md [hunk 5]
```diff
@@ -343,15 +406,174 @@ for (const message of batch.messages) {
...
+## 8. Deploy your Worker
 
 To deploy your Worker, run the following command:
 
 ```sh
 $ npx wrangler deploy
 ```
 
-You have successfully created a Worker which can submit URLs to a queue for crawling and save results to KV.
+You have successfully created a Worker which can submit URLs to a queue for crawling and save results to KV as an [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
```
Comment (maheshwarip) [inline content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: same comment as above, IMO we don't need to specify that we're saving results as an ArrayBuffer

**Comment by maheshwarip** [comment]
couple of minor nits, and one possible typo, but otherwise lgtm! will approve so you can merge anytime once the changes are in

**Review by maheshwarip — APPROVED** [review]

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md (hunk 4 above)
Comment (kodster28) [reply content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: Should be updated by latest commit

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md (hunk 5 above)
Comment (kodster28) [reply content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: Should be updated by latest commit

File: content/queues/tutorials/web-crawler-with-browser-rendering/index.md (hunk 3 above)
Comment (kodster28) [reply content/queues/tutorials/web-crawler-with-browser-rendering/index.md]: Should be updated by latest commit

**Review by Maddy-Cloudflare — APPROVED** [review]

**Pull request merged**